server/app.js

```diff
@@ -8,19 +9,38 @@ const connectDB = require("./db");
 const { join } = require("path");
 const cookieParser = require("cookie-parser");
 const logger = require("morgan");
-const cors = require("cors")
+const cors = require("cors");
+const pkg = require("cloudinary");
+const { v2: cloudinary } = pkg;
 
 const authRouter = require("./routes/auth");
 const userRouter = require("./routes/user");
 
 const { json, urlencoded } = express;
 
-
-connectDB()
+connectDB();
 const app = express();
 const server = http.createServer(app);
 
-app.use(cors())
+const whitelist = ["http://localhost:3000"];
+const corsOptions = {
+  origin: function (origin, callback) {
+    if (whitelist.indexOf(origin) !== -1) {
+      callback(null, true);
+    } else {
+      callback(new Error("Not allowed by CORS"));
+    }
+  },
+  credentials: true,
+};
+
+app.use(cors(corsOptions));
+
+cloudinary.config({
+  cloud_name: process.env.CLOUDINARY_CLOUD,
+  api_key: process.env.CLOUDINARY_API_KEY,
+  api_secret: process.env.CLOUDINARY_API_SECRET,
```

Comment:
Please update the `sample.env` to reflect what environment variables are required to run the back-end. This is great for easy onboarding of a new developer.

Reply:
**Don't expose the API credentials**